src/components/CardEditor/CardEditForm/CardEditFormContent.jsx

```diff
@@ -0,0 +1,250 @@
+  /** if provided, returns an array of strings which are the dataItems to be allowed
+   * on each card
+   * getValidDataItems(card, selectedTimeRange)
+   */
+  getValidDataItems: PropTypes.func,
+  /** an array of dataItem string names to be included on each card
+   * this prop will be ignored if getValidDataItems is defined
+   */
+  dataItems: PropTypes.arrayOf(PropTypes.string),
+};
+
+const defaultProps = {
+  cardJson: {},
+  i18n: {
+    openEditorButton: 'Open JSON editor',
+    cardSize_SMALL: 'Small',
+    cardSize_SMALLWIDE: 'Small wide',
+    cardSize_MEDIUM: 'Medium',
+    cardSize_MEDIUMTHIN: 'Medium thin',
+    cardSize_MEDIUMWIDE: 'Medium wide',
+    cardSize_LARGE: 'Large',
+    cardSize_LARGETHIN: 'Large thin',
+    cardSize_LARGEWIDE: 'Large wide',
+    chartType_BAR: 'Bar',
+    chartType_LINE: 'Line',
+    barChartType_SIMPLE: 'Simple',
+    barChartType_GROUPED: 'Grouped',
+    barChartType_STACKED: 'Stacked',
+    barChartLayout_HORIZONTAL: 'Horizontal',
+    barChartLayout_VERTICAL: 'Vertical',
+    cardTitle: 'Card title',
+    description: 'Description (Optional)',
+    size: 'Size',
+    selectASize: 'Select a size',
+    timeRange: 'Time range',
+    selectATimeRange: 'Select a time range',
+  },
+  getValidDataItems: null,
+  getValidTimeRanges: null,
+  dataItems: [],
+};
+
+export const defaultTimeRangeOptions = {
+  last24Hours: 'Last 24 hrs',
+  last7Days: 'Last 7 days',
+  lastMonth: 'Last month',
+  lastQuarter: 'Last quarter',
+  lastYear: 'Last year',
+  thisWeek: 'This week',
+  thisMonth: 'This month',
+  thisQuarter: 'This quarter',
+  thisYear: 'This year',
+};
+
+/**
+ * Returns card size and dimensions labels
+ * @param {string} size
+ * @param {Object<string>} i18n
+ * @returns {string}
+ */
+export const getCardSizeText = (size, i18n) => {
+  const sizeName = i18n[`cardSize_${size}`];
+  const sizeDimensions = `(${CARD_DIMENSIONS[size].lg.w}x${CARD_DIMENSIONS[size].lg.h})`;
+  return `${sizeName} ${sizeDimensions}`;
+};
+
+const CardEditFormContent = ({
+  cardJson,
+  onChange,
+  i18n,
+  dataItems,
+  getValidDataItems,
+  getValidTimeRanges,
+}) => {
+  const { title, description, size, type } = cardJson;
+  const mergedI18n = { ...defaultProps.i18n, ...i18n };
+  const [selectedDataItems, setSelectedDataItems] = useState([]);
+  const [selectedTimeRange, setSelectedTimeRange] = useState('');
+
+  const baseClassName = `${iotPrefix}--card-edit-form`;
+
+  const validTimeRanges = getValidTimeRanges
+    ? getValidTimeRanges(cardJson, selectedDataItems)
+    : Object.keys(defaultTimeRangeOptions);
+
+  return (
+    <>
+      <div className={`${baseClassName}--input`}>
+        <TextInput
+          id="title"
+          labelText={mergedI18n.cardTitle}
+          light
+          onChange={(evt) => onChange({ ...cardJson, title: evt.target.value })}
+          value={title}
+        />
+      </div>
+      <div className={`${baseClassName}--input`}>
+        <TextArea
+          id="description"
```

Comment:
Should the IDs be merged with the parent id?  ideally IDs should be unique in the DOM but I have  feeling that id="description" would be quite common, should this be something like, {`${this.props.id}_cardeditor_description`}

Reply:
Good observation, I agree. I will update this.